pandas/core/arrays/arrow/array.py

```diff
@@ -2093,7 +2093,7 @@ def _dt_round(
     def _dt_to_pydatetime(self):
         data = self._pa_array.to_pylist()
         if self._dtype.pyarrow_dtype.unit == "ns":
-            data = [ts.to_pydatetime(warn=False) for ts in data]
+            data = [None if isna(ts) else ts.to_pydatetime(warn=False) for ts in data]
```

Comment:
I think could simplify to `ts if ts is None else ...` since it looks like `to_pylist` will return None as the NA value

```
In [12]: pa.array([datetime(2023, 1, 1), None], type=pa.timestamp("ns")).to_pylist()
Out[12]: [Timestamp('2023-01-01 00:00:00'), None]
```

Reply:
good point, done